Sources/Basics/String+Extensions.swift

```diff
@@ -0,0 +1,20 @@
+/*
+ This source file is part of the Swift.org open source project
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import TSCBasic
+
+extension String {
```

Comment:
Great to be adding this, we should probably move some of our more generic string quoting functions etc here as well (not all, of course, as sometimes the quoting is specialized).

Reply:
I actually think this isn't a good idea, we shouldn't add public methods to `String` (or any other type we don't own) in anything that becomes part of libSwiftPM, especially not something generic like this.

Reply:
I think the issue here is public visibility across modules

Reply:
That's a good point, Boris — what I meant was consolidating within SwiftPM, but you're right.  Could we mark this SPI in order to still share within SwiftPM.  Tom is of course right about the underlying issue but since we don't have that, maybe SPI is a reasonable alternative?